### PR TITLE
[DSE] Fixup processing of aliasing expressions

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: tests/test_dse.py [hunk 1]
```diff
@@ -334,18 +334,19 @@ def test_full_shape_after_blocking(self):
         assert Add(*a.symbolic_shape[0].args) == x0_blk_size + 2
         assert Add(*a.symbolic_shape[1].args) == y0_blk_size + 2
         assert Add(*a.symbolic_shape[2].args) == z_size + 2
+
         # Check numerical output
         op0(time_M=1)
         exp = np.copy(u.data[:])
-        u.data_with_halo[:] = 0.
+        u.data_with_halo[:] = 0.5
```
Comment (mloubout) [inline tests/test_dse.py]: why?

File: tests/test_dse.py [hunk 2]
```diff
@@ -373,19 +374,20 @@ def test_contracted_shape_after_blocking(self):
         assert a.halo == ((1, 1), (1, 1))
         assert Add(*a.symbolic_shape[0].args) == y0_blk_size + 2
         assert Add(*a.symbolic_shape[1].args) == z_size + 2
+
         # Check numerical output
         op0(time_M=1)
         exp = np.copy(u.data[:])
-        u.data_with_halo[:] = 0.
+        u.data_with_halo[:] = 0.5
```
Comment (mloubout) [inline tests/test_dse.py]: why?

File: tests/test_dse.py [hunk 3]
```diff
@@ -394,7 +396,7 @@ def test_full_shape_with_subdims(self):
         f = Function(name='f', grid=grid)
         f.data_with_halo[:] = 1.
         u = TimeFunction(name='u', grid=grid, space_order=3)
-        u.data_with_halo[:] = 0.
+        u.data_with_halo[:] = 0.5
```
Comment (mloubout) [inline tests/test_dse.py]: ?

File: devito/targets/common/blocking.py [hunk 4]
```diff
@@ -133,6 +133,9 @@ def fold_blockable_tree(iet, blockinner=True):
                 if len(i) >= 2:
                     groups.append(i)
         for i in groups:
+            # Performance heuristic: give up if fold length > 2:
+            if len(i) > 2:
+                continue
```
Comment (mloubout) [inline devito/targets/common/blocking.py]: Which heuristic?

File: tests/test_dse.py (hunk 2 above)
Comment (FabioLuporini) [reply tests/test_dse.py]: to minimize amount of computation on zeros

File: tests/test_dse.py (hunk 3 above)
Comment (FabioLuporini) [reply tests/test_dse.py]: to minimize amount of computation on zeros

File: devito/targets/common/blocking.py (hunk 4 above)
Comment (FabioLuporini) [reply devito/targets/common/blocking.py]: do not fold if there are too many loop nests to fold

don't worry to much about this code: this whole file will be completely dropped by `hoist-tiling` (WIP branch, you can already take a look), and so the entire concept of folding will disappear

**Comment by FabioLuporini** [comment]
Ready but won't land until I have compelling evidence there are no performance regressions

**Comment by FabioLuporini** [comment]
closing as superseded by #1171

**Pull request closed without merging**